### PR TITLE
nuw, nsw and exact keywords for integral ops

### LINKED ISSUE #52: Integral operations missing 'nuw', 'nsw' and 'exact' keywords
https://github.com/llvm-hs/llvm-hs-pretty/blob/5fea706dda0eff4f819fa762de1735675496872a/src/LLVM/Pretty.hs#L482-L494

### PR DESCRIPTION
**Pull request by SongWithoutWords**
resolves #52

### CODE REVIEW

File: src/LLVM/Pretty.hs [hunk 1]
```diff
@@ -479,17 +479,61 @@ instance PP Terminator where
 
 instance PP Instruction where
   pp = \case
-    Add {..}    -> "add"  <+> ppTyped operand0 `cma` pp operand1 <+> ppInstrMeta metadata
-    Sub {..}    -> "sub"  <+> ppTyped operand0 `cma` pp operand1 <+> ppInstrMeta metadata
-    Mul {..}    -> "mul"  <+> ppTyped operand0 `cma` pp operand1 <+> ppInstrMeta metadata
-    Shl {..}    -> "shl"  <+> ppTyped operand0 `cma` pp operand1 <+> ppInstrMeta metadata
-    AShr {..}   -> "ashr" <+> ppTyped operand0 `cma` pp operand1 <+> ppInstrMeta metadata
-    LShr {..}   -> "lshr" <+> ppTyped operand0 `cma` pp operand1 <+> ppInstrMeta metadata
+    Add {..}    -> "add"
```
Comment (cocreature) [inline src/LLVM/Pretty.hs]: There is quite a bit of duplication going on here. What do you think about factoring this out into two helper functions, one for binary operators that have `nuw` and `nsw` flags and one operators that have an `exact` flag?

Comment (SongWithoutWords) [reply src/LLVM/Pretty.hs]: sounds good!

Comment (cocreature) [reply src/LLVM/Pretty.hs]: I was thinking of a helper that captures the whole instruction and takes as arguments the instruction name, the two operands, the nuw and nsw flags and the metadata. What do you think about that?

Comment (SongWithoutWords) [reply src/LLVM/Pretty.hs]: oh, something like
`intBinOpWithWraparoundFlags :: Doc -> Bool -> Bool -> Operand -> Operand -> InstructionMetadata -> Doc`
and
`intBinOpWithExact :: Doc -> Bool -> Operand -> Operand -> Doc`?

Comment (cocreature) [reply src/LLVM/Pretty.hs]: Exactly!

Comment (SongWithoutWords) [reply src/LLVM/Pretty.hs]: after this most recent change, and if the build and tests pass, then I think it'll be good to go!

**Comment by cocreature** [comment]
Thanks!

**Pull request merged**